roles/container-engine/containerd/templates/config.toml.j2

```diff
@@ -1,78 +1,41 @@
-# persistent data location
-root = "{{ containerd_metadata_root_dir }}"
-# runtime state information
-state = "{{ containerd_state_dir }}"
+version = 2
 
-# Kubernetes doesn't use containerd restart manager.
-disabled_plugins = ["restart"]
-
-[debug]
-  level = "{{ containerd_config.debug.level | default("") }}"
-
-{% if 'grpc' in containerd_config %}
 [grpc]
-{% for param, value in containerd_config.grpc.items() %}
-  {{ param }} = {{ value }}
-{% endfor %}
-{% endif %}
-
-[plugins.linux]
-  shim = "/usr/bin/containerd-shim"
-  runtime = "{{ runc_binary }}"
-
-[plugins.cri]
-  stream_server_address = "127.0.0.1"
-  max_container_log_line_size = {{ containerd_config.max_container_log_line_size }}
-  sandbox_image = "{{ pod_infra_image_repo }}:{{ pod_infra_image_tag }}"
-  systemd_cgroup = {{ containerd_use_systemd_cgroup|lower }}
-
-[plugins.cri.cni]
-  bin_dir = "/opt/cni/bin"
-  conf_dir = "/etc/cni/net.d"
-  conf_template = ""
-
-{% if 'containerd' in containerd_config %}
-[plugins.cri.containerd]
-{% for param, value in containerd_config.containerd.items() %}
-  {{ param }} = "{{ value }}"
-{% endfor %}
-{% endif %}
+  max_recv_message_size = {{ containerd_grpc_max_recv_message_size | default(16777216) }}
+  max_send_message_size = {{ containerd_grpc_max_send_message_size | default(16777216) }}
 
-[plugins.cri.containerd.default_runtime]
-  runtime_type = "{{ containerd_default_runtime.type }}"
-  runtime_engine = "{{ containerd_default_runtime.engine }}"
-  runtime_root = "{{ containerd_default_runtime.root }}"
+[debug]
+  level = "{{ containerd_debug_level | default('info') }}"
 
+[metrics]
+  address = "{{ containerd_metrics_address | default('') }}"
+  grpc_histogram = {{ containerd_metrics_grpc_histogram | default(false) | lower }}
+
+[plugins]
+  [plugins."io.containerd.grpc.v1.cri"]
+    sandbox_image = "{{ pod_infra_image_repo }}:{{ pod_infra_image_tag }}"
+    max_container_log_line_size = {{ containerd_max_container_log_line_size }}
+    [plugins."io.containerd.grpc.v1.cri".containerd]
 {% if kata_containers_enabled %}
-[plugins.cri.containerd.runtimes.kata-qemu]
-  runtime_type = "io.containerd.kata-qemu.v2"
-  [plugins.cri.containerd.runtimes.kata-qemu.options]
-    ConfigPath = "/etc/kata-containers/configuration-qemu.toml"
+      default_runtime_name = "kata-qemu"
```

Comment:
I think default runtime should still be runc (or containerd_default_runtime_name), even if kata is enabled.

Reply:
This line of code is wrapped with some Ansible if-else to determine if Kata containers is enabled or not.

```
{% if kata_containers_enabled %}
      default_runtime_name = "kata-qemu"
{% else %}
```

Reply:
Yes, but kata_containers_enabled==true does not mean it should be the default, only that is should be configured/available in addition to runc. This PR changes previous behaviour here.

Reply:
Yes, you are correct. I've fixed the code to use the value of `containerd_default_runtime` or default to `runc` if not defined.